// mdx-deck/dist/components.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _styledSystem = require('styled-system');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var css = function css(key) {
  return function (props) {
    return props.theme[key];
  };
};

var Heading = _styledComponents2.default.h1([], {
  lineHeight: 1.25
}, _styledSystem.fontSize, _styledSystem.space, _styledSystem.color, css('heading'));
Heading.defaultProps = {
  color: 'heading',
  mt: 3,
  mb: 3
};

var H2 = Heading.withComponent('h2');
var H3 = Heading.withComponent('h3');
var H4 = Heading.withComponent('h4');
var H5 = Heading.withComponent('h5');
var H6 = Heading.withComponent('h6');

var h1 = function h1(props) {
  return _react2.default.createElement(Heading, _extends({}, props, {
    fontSize: [3, 4, 5]
  }));
};

var h2 = function h2(props) {
  return _react2.default.createElement(H2, _extends({}, props, {
    fontSize: [2, 3, 4]
  }));
};

var h3 = function h3(props) {
  return _react2.default.createElement(H3, _extends({}, props, {
    fontSize: [2, 3]
  }));
};

var h4 = function h4(props) {
  return _react2.default.createElement(H4, _extends({}, props, {
    fontSize: [2, 3]
  }));
};

var h5 = function h5(props) {
  return _react2.default.createElement(H5, _extends({}, props, {
    fontSize: [2, 3]
  }));
};

var h6 = function h6(props) {
  return _react2.default.createElement(H6, _extends({}, props, {
    fontSize: [2, 3]
  }));
};

var a = _styledComponents2.default.a([], _styledSystem.color, css('link'));
a.defaultProps = {
  target: '_blank',
  color: 'link'
};

var p = _styledComponents2.default.p([], _styledSystem.fontSize, _styledSystem.space, _styledSystem.color, css('paragraph'));

var ul = _styledComponents2.default.ul([], {
  textAlign: 'left'
}, _styledSystem.fontSize, css('ul'));
ul.defaultProps = {
  fontSize: [2, 3]
};

var ol = _styledComponents2.default.ol([], {
  textAlign: 'left'
}, _styledSystem.fontSize, css('ol'));
ol.defaultProps = {
  fontSize: [2, 3]
};
var li = _styledComponents2.default.li([]);

var blockquote = _styledComponents2.default.blockquote([], {
  textAlign: 'left',
  fontWeight: 'bold'
}, _styledSystem.fontSize, _styledSystem.space, _styledSystem.color, css('quote'));
blockquote.defaultProps = {
  fontSize: [2, 3, 4],
  px: 0,
  mx: 0,
  color: 'quote'
};

var pre = _styledComponents2.default.pre([], function (props) {
  return {
    fontFamily: props.theme.monospace
  };
}, _styledSystem.fontSize, _styledSystem.space, _styledSystem.color, css('pre'));
pre.defaultProps = {
  fontSize: [1, 2],
  m: 0,
  p: 2,
  color: 'pre',
  bg: 'preBackground'
};

var code = _styledComponents2.default.code([], function (props) {
  return {
    fontFamily: props.theme.monospace
  };
}, _styledSystem.fontSize, _styledSystem.space, _styledSystem.color, css('code'));
code.defaultProps = {
  color: 'code',
  bg: 'codeBackground'
};

var img = _styledComponents2.default.img([], {
  maxWidth: '100%',
  height: 'auto'
}, css('image'));

exports.default = {
  h1: h1,
  h2: h2,
  h3: h3,
  h4: h4,
  h5: h5,
  h6: h6,
  a: a,
  p: p,
  blockquote: blockquote,
  ul: ul,
  ol: ol,
  li: li,
  pre: function pre(props) {
    return props.children;
  },
  code: pre,
  inlineCode: code,
  img: img
};
